simple.kt

```diff
@@ -0,0 +1,8 @@
+fun main() = generateSequence { readLine() }
+        .flatMap { it.toLowerCase().splitToSequence(' ') }
+        .filter { !it.isBlank() }
```

Comment:
IMO, it's better to `.filter { it.isNotEmpty() }` or `.filterNot(String::isEmpty)`

Also you may redefine `readLine` as 
```kotlin
val readLine = System.`in`.reader().buffered(65535)::readLine
```
and get "optimized" version for free (~30% boost)

Reply:
Regarding the first part, good point, changed to `it.isNotBlank()` slightly longer but cleaner, I believe `it.isNotEmpty()` wouldn't work in this particular case.

Regarding the second point. This was my initial version, but Kotlin != Java. The current code can be easily set up as a multiplatform Kotlin project and compiled also as JavaScript and native and benchmarked separately, as I am doing in this project:

https://github.com/xemantic/java-2-times-faster-than-c

If I was to provide highly performant Kotlin code, it would look differently and probably might be dependant on JVM. For the simple one I see no good reasons to.

Reply:
> The current code can be easily set up as a multiplatform Kotlin project
Technically - yes, it's possible to compile current version with kotlin-native, but
 - it runs ~ 9 sec vs ~2 sec (jvm)
 - it produces different output
 So, it seems, you'll need different version for each platform in any case

Reply:
> * it runs ~ 9 sec vs ~2 sec (jvm)

Hmm, for me it was something like 1.25s vs 1.36s, in favor of JVM. I was using 1.4.31 compiler, I wonder where does the difference come from?

> * it produces different output

This I haven't checked, in general all these scripts are producing different output because often there are several words with the same number of occurrences. There are no rules for specified for the order of them. This is the reason why all the results are being normalized before comparison.

Reply:
I will try to compile with mimalloc and see if there is any imrovement

Reply:
> * it produces different output
Indeed, `readLine` in case of kotlin native seems to be dropping data :( I don't see any kotlin native alternative for standard input processing which is disappointing.